scripts/arm_aci.py

```diff
@@ -0,0 +1,199 @@
+from azure.mgmt.resource.resources.models import (
+    Deployment,
+    DeploymentProperties,
+    DeploymentMode,
+)
+from azure.mgmt.containerinstance import ContainerInstanceManagementClient
+
+try:
+    HOST_PUB_KEY = open(os.path.expanduser("~/.ssh/id_rsa.pub"), "r").read().replace("\n", "")
+except Exception as e:
+    ...
+
+
+STARTUP_COMMANDS = {
+    "dynamic-agent": lambda args, i: [
+        "apt-get update",
+        "apt-get install -y openssh-server rsync",
+        "sed -i 's/PasswordAuthentication no/PasswordAuthentication yes/g' /etc/ssh/sshd_config",
+        "useradd -m agent",
+        "echo \"agent ALL=(ALL) NOPASSWD: ALL\" >> /etc/sudoers",
+        f'echo "agent:{args.aci_dynamic_agent_password}" | chpasswd',
+        "service ssh restart",
+    ],
+    "static-agent": lambda args, i: [
+        "apt-get install wget",
+        "wget https://gist.github.com/DomAyre/98d3a229870f4947fc99a2aa7ed995b4/raw -O setup_agent.sh",
+        "chmod 777 setup_agent.sh",
+        f"AGENT_NAME={args.deployment_name}-{i} PAT={args.aci_pat} ./setup_agent.sh",
+    ],
+    "dev": lambda args, i: [
+        "apt-get install wget",
+        "wget https://gist.github.com/DomAyre/ea2c07a9cb790bf17da05d4ca1674c8c/raw -O setup_dev.sh",
+        "chmod 777 setup_dev.sh",
+        " ".join([
+            f"MSUSER=\"{args.aci_ms_user}\"",
+            f"GITHUBUSER=\"{args.aci_github_user}\"",
+            f"GITHUBNAME=\"{args.aci_github_name}\"",
+            f"SSHKEYS=\"{args.aci_ssh_keys}\"",
+            "./setup_dev.sh"
+        ]),
+    ],
+}
+
+
+def make_aci_deployment(parser: ArgumentParser) -> Deployment:
+
+    parser.add_argument(
+        "--aci-image",
+        help="The name of the image to deploy in the ACI",
+        type=str,
+        default="ccfmsrc.azurecr.io/ccf/ci:oe-0.18.4-snp",
+    )
+
+    parser.add_argument(
+        "--aci-type",
+        help="The type of ACI to deploy",
+        type=str,
+        choices=STARTUP_COMMANDS.keys(),
+    )
+
+    parser.add_argument(
+        "--aci-pat",
+        help="The PAT to deploy an ACI with",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--aci-ms-user",
+        help="The Microsoft User",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--aci-github-user",
+        help="The Github User who owns a CCF clone to checkout",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--aci-github-name",
+        help="The name to commit with",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--aci-ssh-keys",
+        help="The ssh keys to add to the dev box",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--aci-dynamic-agent-password",
+        help="The password to set on the ACI",
+        type=str,
+    )
+
+    args = parser.parse_args()
+
+    return Deployment(
+        properties=DeploymentProperties(
+            mode=DeploymentMode.INCREMENTAL,
+            parameters={},
+            template={
```

Comment:
This JSON schema should probably become a `.jinja` file eventually, so that it's easier to modify, etc. in isolation.

Reply:
Yeah sounds good, I reckon these two comments could be rolled into a "make arm deployments more flexible" issue